### ISSUE TITLE
The following error occurred attempting to run the project model server process (1.0.0-preview2-003121).

### ISSUE THREAD

**Issue by andez2000** [aspnet/Tooling#629]
The link on https://www.microsoft.com/net/core#windows under link .NET Core for Visual Studio official MSI Installer. still says Preview 2 – “DotNetCore.1.0.0-VS2015Tools.Preview2.exe” Is this correct?

What do I need to get VS2015 working with ASP.NET Core? I’ve installed VS2015 update 3. Then it told me to install the above link so I did. When I create a ASP.NET Core project in VS2015 I get the message once the wizard completes and everything appears in the solution explorer.  I also get this when debugging the project.

The following error occurred attempting to run the project model server process (1.0.0-preview2-003121).

Unable to start the process.

The project model provides intellisense, build and reference information to Visual Studio and without it your experience will be very limited. I rebooted machine after 2015 was installed. I did find the initial SDK installed so I removed it and repaired the setup from “DotNetCore.1.0.0-VS2015Tools.Preview2.exe”.

Any way to diagnose this problem?

Also when I click compile I get:
1>  C:\Program Files\dotnet\dotnet.exe build "E:\Code Workbench\Web\ASP.NET Core\Portal2016.1\src\Portal2016.1" --configuration Debug --no-dependencies

See the attached zip of what I get generated by the project wizard.

This is my development environment at work.  I have had different issues updating Visual Studio 2015 to Update 3 and creating ASP.NET Core projects.

[Portal2016.1.zip](https://github.com/aspnet/Tooling/files/338895/Portal2016.1.zip)

Update

This happens for all .NET Core Project Types - ClassLib, Console, and ASP.NET.

**Comment by radams0x** [comment]
I get something similar: The following error occurred attempting to run the project model server  process (1.0.0-preview2-003121) No connection could be made because the target server actively refused it

**Comment by rmtew** [comment]
I get this as well.

While still on vs update 2, and dot net preview 1, I got it the first time I opened my solution but closing VS and restarting it would make it go away.

Now I consistently get it while opening my solution every time.

**Comment by rmtew** [comment]
Okay, while I could do a non-incremental build, and didn't seem to be able to debug or anything else, I did type "dotnet run" in the console.

```
C:\RMT\VCS\_E\v.m\src\v.m>dotnet run
Project v.m (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
The specified framework 'Microsoft.NETCore.App', version '1.0.0-rc2-3002702' was not found.
  - Check application dependencies and target a framework version installed at:
      C:\Program Files\dotnet\shared\Microsoft.NETCore.App
  - The following versions are installed:
      1.0.0
  - Alternatively, install the framework version '1.0.0-rc2-3002702'.

```
So looking at project.json, most of the versions had been updated, but not:

```
    "Microsoft.NETCore.App": {
      "version": "1.0.0",
      "type": "platform"
    },

```

Which before I changed it to "1.0.0", referred to: "1.0.0-rc2-3002702".  I was directed to look in the shared framework directory of the dotnet installation for the version, which was "1.0.0".  Other than that the remaining problems are version conflicts which I need to manually resolve.

So now I can compile with errors, and debug, and this particular error has gone away for me.

**Comment by barrytang** [comment]
@troydai, can you take a look?

cc @BillHiebert

**Comment by troydai** [comment]
Looks like there are two issues been reported here.

Issue 1, from original post, it seems that VS failed to locate dotnet process to start project model server. I'm not familiar with installer and WTE. I'll leave this to @BillHiebert 

Issue 2, project model server refuse connection for the first start up. Issue is probably caused by `dotnet` local cache initialization. @BillHiebert knows the context.

**Comment by nboisvert** [comment]
@troydai @andez2000 @barrytang Any update on this one ? I'm stuck with the same problem since the release or preview last week ! I'm the only one at my job to get this problem, also the only one to have previously installed every RC released...

**Comment by BillHiebert** [comment]
@nboisvert  What do you have in your c:\program files\dotnet\sdk folder? How many versions of the sdk do you have?

**Comment by nboisvert** [comment]
@BillHiebert I uninstalled all previous RC installed, so the only folder i have in c:\program files\dotnet\sdk is 1.0.0-preview2-003121. The exact error message i get is here

![aspnetcore](https://cloud.githubusercontent.com/assets/3764932/16628477/e9116a86-437f-11e6-9e2c-d73e68f369a8.JPG)

I get this on new project creation AND on opening existing .net core project created by one of my colleague.
The new project created then has no references and cannot be started since VS doesn't recognize the project.

thanks for your time

**Comment by BillHiebert** [comment]
@nboisvert that error generally means the dotnet.exe failed to start. You can run the project model server from the command line as follows.
1. Start notepad.exe and get its process id from taskmgr. This is only needed since the project model server needs a parent process (it terminates itself when the parent process ends). Leave it running
2. cd to your project folder
3. run the following command line. The port can be any unused port (the one below is probably ok). replace the -host-pid with notepads process id from step 1. --host-name can be anything.
`"c:\program files\dotnet\dotnet.exe" projectmodel-server --port 1348 --host-pid 2816 --host-name abcdefg`

Do you see any errors in the output window?

**Comment by nboisvert** [comment]
@BillHiebert No errors, process started correctly. The problemes really seems to be VS unable to start it. Tried uninstall VS and reinstall it this morning with no success.

**Comment by BillHiebert** [comment]
@nboisvert Are there any other instances of dotnet.exe on your path? From a command prompt type "where dotnet.exe".

**Comment by andez2000** [comment]
@BillHiebert  For me everything you have said I have tried.  I only have one dotnet.exe in C:\Program Files\dotnet\dotnet.exe with "where dotnet.exe"

I tried the notepad.exe with dotnet.exe and that works ok.  Just getting the error running from Visual Studio 2015 as per original post and @nboisvert screenshot when I load the project.

**Comment by nboisvert** [comment]
@BillHiebert @andez2000 Finally found out what the problem was (for me). I had a test framework extension called 'JustMock' installed on my Visual Studio. Disabling the extension fixed my problem.
They'll have to fix this on their side. Thanks @BillHiebert for your time on this one !

**Comment by andez2000** [comment]
@nboisvert excellent hunting around.  How did you figure that?  I have JustMock too!!!  I've uninstalled and it works now. Have you raised this with Telerik?  Yep thanks @BillHiebert

**Comment by BillHiebert** [comment]
@nboisvert. Thanks for tracking this down.  I was completely out of ideas to try next. I've never seen a case where it succeeds from the command line but does not work from VS.

**Comment by brentdavid2008** [comment]
Also had the same thing happen.  I sent in a Telerik bug report.  will wait and see what they say.

**Comment by ToddThomson** [comment]
i also get the dialog box reporting the project model server process error. It only happens periodically when I open a large project first thing in the morning. I close VS and then re-open the project. The 2nd time is fine.

**Comment by mlorbetske** [comment]
TFS Bug#241363

**Comment by mlorbetske** [comment]
Since this has been tracked down to being the JustMock tool, closing the issue

**Comment by ToddThomson** [comment]
I still get this problem the first time I run VS2015 Community in the morning. I do not have JustMock installed.

**Comment by radams0x** [comment]
I also get the same and I don't have JustMock

**Comment by mlorbetske** [comment]
Reopened for further investigation, sorry about that

**Comment by ales004** [comment]
In my case, I got an error like the one described [here](https://social.msdn.microsoft.com/Forums/vstudio/en-US/c93830b5-d5c6-4f21-afd5-4f7480a62b22/no-connection-could-be-made-because-the-target-machine-actively-refused-it-ffff12700153216?forum=visualstudiogeneral)

In my case the problem seemed to be related to the fact that the build was failing. Is it ok that donet restore is related to the success of the build procedure?

**Comment by chrislyse** [comment]
I also get this error on a fresh windows installation.  Installed Sql server 2016, Visual Studio 2015 Update 3.  When installing the VS Tools i get the error (like everyone in my office also gets: http://www.ryadel.com/en/microsoft-net-core-1-0-0-vs2015-tooling-preview-update-2-fails-install-error-0x81f4000-fix/

I think try to create a demo project, and i can not run it due to the same error as listed on this page...I did not install anything else and i have no idea what the issue is.  I did not have these issues when trying the RC's in the past on my work pc.  We're off to a shaky start.

**Comment by barrytang** [comment]
Hi @chrislyse, the latest update to VS 2015 Update 3 caused an issue to the .NET Core Tooling installer. This should be fixed now. Could you please try installing from https://www.microsoft.com/net/core#windows again? Thanks!

**Comment by daramire** [comment]
I get this as well. 

![vs2015 error](https://cloud.githubusercontent.com/assets/7254799/17782604/0916b616-653a-11e6-919c-a1754b71eb41.png)

**Comment by SliceofCheese** [comment]
It seems that your error is similar to mine but gives you a different error: I have
![baderror](https://cloud.githubusercontent.com/assets/21269526/18021393/136f62da-6bb5-11e6-8966-a0fca0e4c7fe.png)

@daramire, I know it might seem pointless to ask now, but can you verify what process 7840 was? Perhaps that can help with determining what's happening here, as I find your error is a bit more straightforward than mine. At least maybe we can get someone through this. Maybe then we can get others working as well.

**Comment by daramire** [comment]
@SliceofCheese, I gave up and ended up doing a clean install of windows 10, then a new install of vs2015 community. The error is gone now.

**Comment by thtp** [comment]
Same here with everything latest:
![image](https://cloud.githubusercontent.com/assets/6025469/18281676/6ae4da2c-7466-11e6-80be-ac381fd30d57.png)

**Comment by cybage-sumedh** [comment]
I am experiencing exactly same error. Nothing has helped as yet.

**Comment by t0xicDen** [comment]
![bug](https://cloud.githubusercontent.com/assets/7147684/18327721/7a323da0-7554-11e6-8be2-c71384b32f74.png)

I got the same error after updating all packages and installing latest SDK and Runtime versions.

**Comment by pealmeid** [comment]
Having the same error since upgrading to runtime version 003590.

![vs](https://cloud.githubusercontent.com/assets/6384219/18422462/bf0054f6-7871-11e6-8d94-bce2fc99b958.jpg)

No error after deleting 003590 and reverting to the previous version (in my case, 003246).
Visual Studio 14.0.25425.01 Update 3

**Comment by pealmeid** [comment]
Apparently `projectmodel-server` was removed from the `dotnet` cli commands:

**https://github.com/dotnet/cli/pull/3752 - Remove Project Model Server**

What is the recommended way to proceed in this case? Thanks in advance.

**Comment by pealmeid** [comment]
Working for me after latest update (version 1.0.0-preview3-003599). Thanks.

**Comment by SliceofCheese** [comment]
Is everyone that's having this issue on Windows 8.1? I tested this problem on another computer, and this works perfectly fine on my laptop which is Windows 7. I don't have any other Windows 8.1 machines, and I've tried uninstalling this and reinstalling it on the windows 8 machine. 

This worked fine on 7 the first try.

**Comment by t0xicDen** [comment]
@SliceofCheese , In my case it's not working on Windows 7.

**Comment by ace26** [comment]
Yup I second @pealmeid Solution.

I had the same issues as everyone here so I decided to install the latest VS 2015 Tooling preview from the dotnet site and still that didn't solve the problem until I installed the latest .NET Core SDK for Windows and that did it!

**Comment by gravity00** [comment]
Not sure if this helps, but to solve this tooling problems (`1.0.0-preview2-003121`) I typically do the following steps:

1. Close solution (or close Visual Studio);
2. Delete all `project.lock.json` files;
3. `dotnet restore --no-cache` in the root directory;
4. Reopen solution;
5. Rebuild all;

Eventually I may manually delete all the `bin` or `obj` folders. The first step is very important to prevent Visual Studio to automatically restore (which normally keeps failing).

**Comment by marchy** [comment]
Our Visual Studio 2015 just got hosed with this error after installing the 2017 RC that was just released. Opening our project in 2015 now throws the following error:

"The following error occurred attempting to run the project model server process (1.0.0-preview3-004056)

Unable to start the process.

The project model server process provides intellisense, build and reference information to Visual Studio and without it your experience will very limited."

Additionally the solution above doesn't work because "dotnet restore --no-cache" fails with:
an "error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0-preview3-004056\Extensions\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk."

How do we get un-mangled??

**Comment by alexavrutin** [comment]
I am experiencing the same condition as @marchy. Just installed VS 2017 RC.

**Comment by jculverwell** [comment]
I am also experiencing the same condition as @marchy.  Also Just installed VS 2017 RC.

**Comment by mehalick** [comment]
Same for me, just installed VS 2017 RC as well...

**Comment by SuperJMN** [comment]
This happened to me just after the installation of Visual Studio 2017 RC.

**Comment by kirill-chilingarashvili** [comment]
I had the same issue,
Found 4 SDK versions on the machine,
uninstalled all + VS tooling, installed latest from https://www.microsoft.com/net/core
now it works

**Comment by SimonOrdo** [comment]
My VS2015 installation was also hosed after the RC2017 install with the same error message as above.  When trying to follow @gravity00 suggestion about a clean restore, above, I got this error message from the dotnet restore command:

> error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0-preview3-004056\Extensions\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

Adding the global.json as per @mehalick , above, worked.  No more errors and intellisense is working again.

**Comment by mariuszjamro** [comment]
I had similar issue with "an attempt was made to access a socket in a way forbidden by its access permissions" message.

In my case is was caused by antivirus software, after uninstalling the issue is gone..

**Comment by erikpowa** [comment]
@thtp  @t0xicDen @pealmeid 
I disabled **Windows Firewall** service and that error is gone. I suggest to try messing around in the Inbound/Outbound rules.

**Comment by atrauzzi** [comment]
Solution from @mehalick works!  Anyone else think there's way too many places to update this kind of metadata?  Seems like it should all be contained in the `project.json`/`csproj`/`xproj` file.  Not strewn about everywhere in my solutions which is VS specific...

**Comment by davidfowl** [comment]
> Anyone else think there's way too many places to update this kind of metadata?

The `global.json` file is the only place you can change the SDK version. It's not VS specific at all. It also affects the version used on the command line.

**Comment by atrauzzi** [comment]
Would there be any sense in just having those kinds of settings live in the project?

**Comment by davidfowl** [comment]
> Would there be any sense in just having those kinds of settings live in the project?

What happens when you have more than a single project? If you want, you can put a `global.json` in your project but that quickly falls over when you have more than a single project.

**Comment by atrauzzi** [comment]
Would that not be the point?  Couldn't each project legitimately prefer a different SDK?

And what I mean by "in project" is, get rid of `global.json` and move this setting into the upcoming csproj stuff.  Then there'd be one less nuance to track.

**Comment by davidfowl** [comment]
That's very rare and projects within the same solution are usually interleaved. Global.json in the future may be renamed to something else (dotnet.json) but it'll have the same purpose, pick the version of the CLI itself. That has to be known befor even looking at the project file. By then it's too late.

**Comment by atrauzzi** [comment]
I'm not so sure it is rare.

Just an off the wall thought, but couldn't you just have fields in the project file that are not intended for msbuild?  Multiple tools can read that file - please tell me there isn't a 1:1 correlation of configuration files to departments.  ;)

But further to it all.. When I add any community project to my solution so that I can debug through source, there's a risk of this coming up.
And I have to say, it's early days but since figuring out how to do it, I definitely see myself adding package projects to solutions more often.

**Comment by davidfowl** [comment]
> Just an off the wall thought, but couldn't you just have fields in the project file that are not intended for msbuild?

Sure, it's a mega hack though and we shouldn't do that. To know what version of msbuild to launch, you shouldn't have to read an msbuild file 😄 .

**Comment by atrauzzi** [comment]
Dunno, doesn't seem so bad put that way either.  It isn't parsing in entirety.  Just looking for specific information.  The fact that they are contained in the same file would be immaterial to the tooling. You're subject to the same issues either way as well.

**Comment by AdrienTorris** [comment]
Same as @mehalick. His solution has work for me too.

**Comment by pholly** [comment]
I had the same error opening a solution (opennidict-core) in VS 2015 as @marchy:
"The following error occurred attempting to run the project model server process (1.0.0-preview3-004056)

Unable to start the process.

The project model server process provides intellisense, build and reference information to Visual Studio and without it your experience will very limited."

I uninstalled Visual Studio 2017 RC and restarted and that fixed the issue.

**Comment by Pabreetzio** [comment]
I couldn't add a global.json for my solution, or didn't know how (I tried). The post from kirill-chilingarashvili helped me figure out how to get past the error. I  opened Add or Remove Programs, scrolled down to Microsoft .Net Core and saw two SDKs. I removed the newer one, SDK 1.0.3, and was left with just SDK 1.0.0 Preview. When I attempted to open the project I got a different error about a dll being missing. I'm guessing it was removed as part of the uninstall of the newer sdk or something. Next I just clicked Modify on the remaining .Net Core SDK from Add or Remove Programs and clicked repair. After it was done repairing I no longer got the error.

**Comment by skaa** [comment]
Same error opening solution in VS 2015 with latest SDK `dotnet-dev-win-x64.1.0.0-rc4-004771.exe` and tooling `DotNetCore.1.0.1-VS2015Tools.Preview2.0.3.exe`:
> The following error occurred attempting to run the project model server process (1.0.0-rc4-004771). 
> Unable to start the process. No executable found matching command "dotnet-projectmodel-server"

`dotnet restore` print error:
>  error MSB4019: The imported project "C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

xproj file has lines:
`<VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>`
and
`<Import Project="$(VSToolsPath)\DotNet\Microsoft.DotNet.Props" Condition="'$(VSToolsPath)' != ''" />`
Directory `C:\Program Files\dotnet\sdk\1.0.0-rc4-004771\Microsoft` doesnt exists.

**Comment by karoui28** [comment]
i am using a VPN i disactivated it and disconnected from WIFI and the error was gone and everything works fine

**Comment by paulh-rnd** [comment]
For me a similar error came down to a stray reference in my Path environment variable which pointed back to a prior .Net Core installation at **C:\Program Files (x86)\dotnet**, which didn't seem to get cleaned up when I uninstalled.  I removed the old path (and reinstalled .Net Core v1.1 SDK for good measure) and let the correct path at **C:\Program Files\dotnet** take over.

**Comment by arntj** [comment]
TLDR;
This problem comes up if you try to open a project.json-based solution in VS2015 after having installed a more recent .NET Core SDK version (e.g. by installing VS2017). Fix it by migrating your solution to the newest .NET Core, or by adding a reference to SDK version 1.0.0-preview2.1-3177 or older in the global.json. (Make sure this .NET SDK is actually installed on your computer first.)

```
{
  "sdk": {
    "version": "1.0.0-preview2-1-003177"
  }
}
```

**Comment by anhmaker** [comment]
About "sdk"-"version" read here:
https://docs.microsoft.com/en-us/dotnet/articles/core/tools/global-json
